### PR TITLE
feat: add saved LLM settings test endpoint

### PR DESCRIPTION
**Pull request by Sdongmaker**
## Summary
- add `POST /api/v1/system/llm-test` to validate saved LLM settings with a minimal real invoke
- sanitize LLM test errors, freeze provider/model/api key/base URL snapshot per request, and reject empty/non-text responses
- add endpoint and helper tests covering disabled agent, missing config, timeout, sanitization, empty reply, and snapshot behavior

## Test Plan
- [x] `uv run --python 3.11 --isolated --with pytest --with fastapi --with pydantic --with pydantic-settings --with starlette --with python-multipart --with anyio --with python-dotenv pytest tests/test_system_llm_test.py tests/test_llm_helper_testcall.py tests/test_system_nettest.py -q`

### CODE REVIEW

File: tests/test_llm_helper_testcall.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        PROXY_HOST=None,
+    ),
+)
+_stub_module("app.log", logger=_DummyLogger())
+
+module_path = "/Users/sdongmaker/VScode/MoviePilot/app/helper/llm.py"
+spec = importlib.util.spec_from_file_location("test_llm_module", module_path)
+llm_module = importlib.util.module_from_spec(spec)
+assert spec and spec.loader
+spec.loader.exec_module(llm_module)
```
[the remark this answers is not in the document]
Comment (Sdongmaker) [reply tests/test_llm_helper_testcall.py]: Fixed in a2fdfa86.

The absolute path is gone. I kept the isolated module loading pattern, but switched it to a repository-relative path built from `Path(__file__).resolve()` so the test remains portable across CI and other machines.

I did not replace it with `import app.helper.llm` directly because `app.helper.__init__` imports `cloudflare`, which pulls in `pyquery`. In the isolated test environment that introduces an unrelated dependency and breaks the helper test setup. The current version keeps the test self-contained while removing the machine-specific path.
